candidate_models/base_models/__init__.py

```diff
@@ -446,8 +445,8 @@ def __init__(self):
 
     def __getitem__(self, basemodel_identifier):
         if basemodel_identifier in self._accessed_base_models:
-            raise ValueError(f"can retrieve each base model only once per session due to possible hook clashes - "
-                             f"{basemodel_identifier} has already been retrieved")
+            model = super(BaseModelPool, self).__getitem__(basemodel_identifier)
+            model.unregister_hooks()
```

Comment:
If there were hooks before, I would issue a warning here. Could e.g. have `unregister_hooks` return all the hooks that were unregistered and then warn (user warning or logger warning) if there were any.